XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -229,15 +229,25 @@ private void CreateNonBonded()
             XenAPI.Network network = PopulateNewNetworkObj();
 
             PIF nic;
-            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            if (pageNetworkDetails.CreateVlanOnSriovNetwork)
             {
-                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];
-                nic = Pool != null ? Pool.Connection.Resolve(Pool.Connection.Resolve(sriovLogicalPif).logical_PIF) : Host.Connection.Resolve(Host.Connection.Resolve(sriovLogicalPif).logical_PIF);
+                if (pageNetworkDetails.SelectedHostNic == null || pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of == null || pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of.Count == 0)

```

Comment:
Here we should use the IsSriovPhysicalPIF() function introduced in PR #1982. We can do it in a separate PR after that one is merged.